js/featured-article-image.js

```diff
@@ -0,0 +1,182 @@
+// any global scripts needed for theme goes here
+;
+(function($) {
+
+    'use strict';
+	var __ = wp.i18n.__;
+	var cropOptions = ( attachment, controller ) => {
+		var control = controller.get( 'control' );
+
+		console.log( attachment );
+		console.log( control );
+
+		var flexWidth = !! parseInt( control.params.flex_width, 10 );
+		var flexHeight = !! parseInt( control.params.flex_height, 10 );
+
+		var realWidth = attachment.get( 'width' );
+		var realHeight = attachment.get( 'height' );
+
+		let xInit = parseInt( control.params.width, 10 );
+		let yInit = parseInt( control.params.height, 10 );
+
+		var ratio = xInit / yInit;
+
+		controller.set( 'canSkipCrop', false );
+
+		var xImg = xInit;
+		var yImg = yInit;
+
+		if ( realWidth / realHeight > ratio ) {
+			if ( yImg > realHeight ) {
+				yImg = realHeight;
+			}
+			yInit = yImg;
+			xInit = yInit * ratio;
+		} else {
+			if ( xImg > realWidth ) {
+				xImg = realWidth;
+			}
+			xInit = xImg;
+			yInit = xInit / ratio;
+		}
+
+		var x1 = ( realWidth - xInit ) / 2;
+		var y1 = ( realHeight - yInit ) / 2;
+
+		var imgSelectOptions = {
+			handles: true,
+			keys: true,
+			instance: true,
+			persistent: true,
+			imageWidth: realWidth,
+			imageHeight: realHeight,
+			x1,
+			y1,
+			x2: xInit + x1,
+			y2: yInit + y1,
+			aspectRatio: '40:21',
+		};
+		return imgSelectOptions;
+	};
+
+	/**
+	 * Set the attachment image.
+	 * @param {object} image Image object after cropping.
+	 */
+	function setAttachmentImage( image ) {
+		console.log( image );
+	}
+
+	var cropControl = {
```

Comment:
This is where params are set. Flex width and flex height allow the image to maintain a crop if smaller or larger.

Width/Height is the desired width/height for the image.